Microsoft.Toolkit.Uwp.UI.Controls/Carousel/CarouselAutomationPeer.cs

```diff
@@ -0,0 +1,148 @@
+        /// </param>
+        public CarouselAutomationPeer(Carousel owner)
+            : base(owner)
+        {
+        }
+
+        /// <summary>Gets a value indicating whether the Microsoft UI Automation provider allows more than one child element to be selected concurrently.</summary>
+        /// <returns>True if multiple selection is allowed; otherwise, false.</returns>
+        public bool CanSelectMultiple => false;
+
+        /// <summary>Gets a value indicating whether the UI Automation provider requires at least one child element to be selected.</summary>
+        /// <returns>True if selection is required; otherwise, false.</returns>
+        public bool IsSelectionRequired => false;
+
+        private Carousel OwningCarousel
+        {
+            get
+            {
+                return Owner as Carousel;
+            }
+        }
+
+        /// <summary>Retrieves a UI Automation provider for each child element that is selected.</summary>
+        /// <returns>An array of UI Automation providers.</returns>
+        public IRawElementProviderSimple[] GetSelection()
+        {
+            return OwningCarousel.ContainerFromItem(this.OwningCarousel.SelectedItem) is CarouselItem selectedCarouselItem
+                       ? new[] { this.ProviderFromPeer(FromElement(selectedCarouselItem)) }
+                       : new IRawElementProviderSimple[] { };
+        }
+
+        /// <summary>
+        /// Gets the control type for the element that is associated with the UI Automation peer.
+        /// </summary>
+        /// <returns>The control type.</returns>
+        protected override AutomationControlType GetAutomationControlTypeCore()
+        {
+            return AutomationControlType.List;
+        }
+
+        /// <summary>
+        /// Called by GetClassName that gets a human readable name that, in addition to AutomationControlType,
+        /// differentiates the control represented by this AutomationPeer.
+        /// </summary>
+        /// <returns>The string that contains the name.</returns>
+        protected override string GetClassNameCore()
+        {
+            return Owner.GetType().Name;
+        }
+
+        /// <summary>
+        /// Called by GetName.
+        /// </summary>
+        /// <returns>
+        /// Returns the first of these that is not null or empty:
+        /// - Value returned by the base implementation
+        /// - Name of the owning Carousel
+        /// - Carousel class name
+        /// </returns>
+        protected override string GetNameCore()
+        {
+            string name = string.Empty;
+
+            if (this.OwningCarousel != null)
+            {
+                name = this.OwningCarousel.Name;
+            }
+
+            if (string.IsNullOrEmpty(name))
+            {
+                name = base.GetNameCore();
+            }
+
+            if (string.IsNullOrEmpty(name))
+            {
+                name = this.GetClassName();
```

Comment:
Why are we returning the class name? Narrator should already announce the class/type of the control, having the name of the control be the same makes it more confusing for users relying on that.

Reply:
Interesting. This is the standard that I've seen across a lot of Windows controls and even based this one off the DataGrid within the toolkit. 

Do agree though that this could be confusing.

Reply:
From what I have seen, WinUI 2 controls (mostly) don't do that, they try to get the a name from AutomationProperties.Name or their content; if that fails they return an empty string.

Reply:
This is a good compromise. I'll do some testing with the inspect tool. I don't know what the outcome would be for the name of the control if it is null or an empty string from the automation peer.

Reply:
Updated this to return the name of the control if provided, otherwise it will fallback to the base implementation.